internal/plugin/index.go

```diff
@@ -101,12 +121,11 @@ func (in Index) Validate() error {
 		if entry.Type != "" && entry.Name != "" && entry.Version != "" {
 			key := strings.Join([]string{string(entry.Type), entry.Name, entry.Version}, ";")
 			firstEntryIdx, found := entriesByKey[key]
-			if !found {
-				entriesByKey[key] = currentIdx
-				continue
+			if found {
```

Comment:
**nit** hm.. so the logic is not consistent with the approach in 110 line, tbh having `if found` and then don't have the `continue` was a bit misleading for me as I was thinking why on error we want to proceed further, and then I realized that there is a map, so it's not a problem at the end however was not readable for me at first look.

Reply:
Yes, the problem is that because of the consistent approach there was a bug 😄 we need to assign this in both cases:
```go
entriesByKey[key] = currentIdx
``` 

I hope this is clearer now:

```go

		if entry.Type != "" && entry.Name != "" && entry.Version != "" {
			// check if we have a duplicate entry
			key := strings.Join([]string{string(entry.Type), entry.Name, entry.Version}, ";")
			firstEntryIdx, alreadyExist := entriesByKey[key]
			if alreadyExist {
				// duplicate, return error
				entryIssues = multierror.Append(entryIssues, fmt.Errorf("conflicts with the %s entry as both have the same type, name, and version", humanize.Ordinal(firstEntryIdx)))
			}
			
			entriesByKey[key] = currentIdx
		}
```

There are two alternative options (copying the assignment, or moving it to the top, before checking `if found`), but honestly I don't like them, so hope the comments and var name change improves the readability

Reply:
Yes, the problem is that because of the consistent approach there was a bug 😄 we need to assign this in both cases:
```go
entriesByKey[key] = currentIdx
``` 

If you go back to the previous form, then when running the index_test.go, you'll get 5 errors instead of 8, as these lines are skipped:

```go
err := entryIssues.ErrorOrNil()
		if err != nil {
			issues = multierror.Append(issues, fmt.Errorf("entries[%d]: %s", currentIdx, stringx.IndentAfterLine(err.Error(), 1, "\t")))
		}
```

I hope this is clearer now:

```go

		if entry.Type != "" && entry.Name != "" && entry.Version != "" {
			// check if we have a duplicate entry
			key := strings.Join([]string{string(entry.Type), entry.Name, entry.Version}, ";")
			firstEntryIdx, alreadyExist := entriesByKey[key]
			if alreadyExist {
				// duplicate, return error
				entryIssues = multierror.Append(entryIssues, fmt.Errorf("conflicts with the %s entry as both have the same type, name, and version", humanize.Ordinal(firstEntryIdx)))
			}
			
			entriesByKey[key] = currentIdx
		}
```

There are two alternative options (copying the assignment, or moving it to the top, before checking `if found`), but honestly I don't like them, so hope the comments and var name change improves the readability